enterprise/backend/src/metabase_enterprise/workspaces/merge.clj

```diff
@@ -19,27 +27,66 @@
 
   ;; Assume validation has alr
   ;; Ensure the hook
-  (let [*tx-id (atom (:global_id ws-transform))]
-    (if @*tx-id
-      (transforms.api/update-transform! @*tx-id (select-keys ws-transform [:name :description :source :target]))
-      ;; TODO need to extract the create-transform function
-      (reset! *tx-id nil #_(:id (transforms.api/create-transform (select-keys ws-transform [:name :description :source :target])))))
+  (let [{:keys [error] :as result}
+        (cond (and global_id archived_at)
+              (merge
+               {:op :delete :global_id global_id :ref_id ref_id}
+               (try
+                 (transforms.api/delete-transform! (t2/select-one :model/Transform :id global_id))
+                 nil
+                 (catch Throwable e
+                   {:error e})))
```

Comment:
nit: Adds a lot of noise having the exact same try-catch boilerplate around each call. Consider using a macro.